staging/src/k8s.io/client-go/dynamic/client.go

```diff
@@ -151,9 +151,18 @@ func (rc *ResourceClient) List(opts metav1.ListOptions) (runtime.Object, error)
 	if parameterEncoder == nil {
 		parameterEncoder = defaultParameterEncoder
 	}
+	resourceName := ""
```

Comment:
Make this block of code a private method of `ResourceClient` to avoid code duplication.

Reply:
Updated. I also put a if logic into `Create()` instead of having two Create functions.